src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -2439,7 +2439,10 @@ private void ClassifyUpdate(TypeDeclarationSyntax oldNode, TypeDeclarationSyntax
                     return;
                 }
 
-                if (!SyntaxFactory.AreEquivalent(oldNode.Modifiers, newNode.Modifiers))
+                // Adding or removing unsafe modifier while debugging shouldn't be an issue.
+                var oldNodeModifiersWithoutUnsafe = oldNode.Modifiers.Where(m => !m.IsKind(SyntaxKind.UnsafeKeyword)).ToSyntaxTokenList();
+                var newNodeModifiersWithoutUnsafe = newNode.Modifiers.Where(m => !m.IsKind(SyntaxKind.UnsafeKeyword)).ToSyntaxTokenList();
```

Comment:
Since we are changing this it would also be nice if we did not report Rude Edits when the modifiers are just reordered (in addition to ignoring `unsafe` differences).

Reply:
I'd also suggest to extract this logic to a helper since we need to apply it to other members that can have `unsafe` modifier:

```C#
unsafe delegate void D(int* x);

class C
{
    unsafe class N { }
    public unsafe event Action<int*[]> A { add { } remove { } }
    unsafe int* F() => null;
    unsafe int* X;
    unsafe int* Y { get; }
}
```

---
In reply to: [506661782](https://github.com/dotnet/roslyn/pull/48629#discussion_r506661782) [](ancestors = 506661782)